api/spec/src/entitlements/entitlements.tsp

```diff
@@ -0,0 +1,321 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi3";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.OpenAPI;
+
+// TODO: does this have to be in a separate namespace?
+namespace OpenMeter.Entitlements;
+
+/**
+ * Order by options for entitlements.
+ */
+@friendlyName("EntitlementOrderBy")
+enum EntitlementOrderBy {
+  #suppress "@openmeter/api-spec/casing" "Use existing values"
+  CreatedAt: "createdAt",
+  #suppress "@openmeter/api-spec/casing" "Use existing values"
+  UpdatedAt: "updatedAt",
+}
+
+@route("/api/v1/entitlements")
+@tag("Entitlements (Experimental)")
+interface Entitlements {
+  /**
+   * List all entitlements for all the subjects and features. This endpoint is intended for administrative purposes only.
+   * To fetch the entitlements of a specific subject please use the /api/v1/subjects/{subjectKeyOrID}/entitlements endpoint.
+   * If page is provided that takes precedence and the paginated response is returned.
+   */
+  @get
+  @operationId("listEntitlements")
+  @summary("List all entitlements")
+  list(
+    /**
+     * Filtering by multiple features.
+     *
+     * Usage: `?feature=feature-1&feature=feature-2`
+     */
+    @query feature?: string[],
+
+    /**
+     * Filtering by multiple subjects.
+     *
+     * Usage: `?subject=customer-1&subject=customer-2`
+     */
+    @query subject?: string[],
+
+    /**
+     * Filtering by multiple entitlement types.
+     *
+     * Usage: `?entitlementType=metered&entitlementType=boolean`
+     */
+    @query entitlementType?: EntitlementType[],
+
+    ...OpenMeter.QueryPagination,
+    ...OpenMeter.QueryLimitOffset,
+    ...OpenMeter.QueryOrdering<EntitlementOrderBy>,
+  ): Entitlement[] | OpenMeter.PaginatedResponse<Entitlement> | OpenMeter.CommonErrors;
```

Comment:
Is this for backwards compatibility reasons (paginated vs non-paginated response)?

Can you please make a note if so?

Reply:
It is, in general everything tries to be as close to the current version as possible. I figured if I were to refactor/rename anything in this PR it would be terribly hard to validate that everything is as expected.